packages/orm/src/relations/ManyToOneReference.ts

```diff
@@ -276,8 +276,17 @@ export class ManyToOneReferenceImpl<T extends Entity, U extends Entity, N extend
     }, otherFieldName: ${this.otherFieldName}, id: ${this.id})`;
   }
 
+  /** Removes pending-hard-delete or soft-deleted entities, unless explicitly asked for. */
   private filterDeleted(entity: U | N, opts?: { withDeleted?: boolean }): U | N {
-    return opts?.withDeleted === true || entity === undefined || !entity.isDeletedEntity ? entity : (undefined as N);
+    if (
+      entity &&
+      (!opts || !opts.withDeleted) &&
+      (entity.isDeletedEntity ||
+        (this.#otherMeta.timestampFields.deletedAt && (entity as any)[this.#otherMeta.timestampFields.deletedAt]))
```

Comment:
imo this should just be codegenned as `isSoftDeletedEntity`.  Since it's a property it'd be undefined for entities without soft deletedness anyways

Reply:
Nice! Good call, will update.